`make test-cpp` hard codes the eigen include dir to `/usr/include/eigen3`

As a result of `tests/Makefile` hardcoding the eigen3 include directory to `/usr/include/eigen3`, if a user follows the instructions given at https://the-walrus.readthedocs.io/en/latest/installing.html#installation.

and runs `make test-cpp`, they will see an error of the form:
```
make -C tests clean
make[1]: Entering directory '/home/poulson/Source/thewalrus/tests'
rm -rf *~ cpptests *.out *.o *.so *.pyc *.mod
make[1]: Leaving directory '/home/poulson/Source/thewalrus/tests'
echo "Going to compile C++ tests"
Going to compile C++ tests
make -C tests cpptests
make[1]: Entering directory '/home/poulson/Source/thewalrus/tests'
g++ gtest_main.cpp -std=c++11 -O3 -g -Wall -fopenmp -fPIC -I/usr/include -I../include -I/usr/include/eigen3 -I/include -march=native -c
g++ libwalrus_unittest.cpp -std=c++11 -O3 -g -Wall -fopenmp -fPIC -I/usr/include -I../include -I/usr/include/eigen3 -I/include -march=native -c
In file included from ../include/libwalrus.hpp:16,
                 from libwalrus_unittest.cpp:18:
../include/eigenvalue_hafnian.hpp:33:10: fatal error: Eigen/Eigenvalues: No such file or directory
 #include <Eigen/Eigenvalues>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:28: libwalrus_unittest.o] Error 1
make[1]: Leaving directory '/home/poulson/Source/thewalrus/tests'
make: *** [Makefile:55: test-cpp] Error 2
```

The preferred approach would seem to be to use the `EIGEN_INCLUDE_DIR` environment variable if it exists, and to fall back to `/usr/include/eigen3` otherwise.

--- a/examples/Makefile
+++ b/examples/Makefile
@@ -11,15 +11,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-EIGEN_INCLUDE_DIR=/usr/include/eigen3
+EIGEN_INCLUDE_DIR?=/usr/include/eigen3
 
 CC=g++
 CFLAGS=-std=c++11 -O3 -Wall -I/usr/include -I../include -I$(EIGEN_INCLUDE_DIR) -fopenmp -march=native
 LFLAGS=-L/usr/lib -lm -fopenmp
 
-all: example-f90 example-cpp
+all: example-cpp
 
-example.o: example.cpp libwalrus.hpp
+example.o: example.cpp
 	$(CC) $^ $(CFLAGS) -c
 
 example-cpp: example.o

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -11,16 +11,13 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-EIGEN_INCLUDE_DIR=/usr/include/eigen3
-#C_INCLUDE_PATH = /opt/local/include/libomp/
-#C_LIBRARY_PATH = /opt/local/lib/libomp/
-#LD_LIBRARY_PATH = /opt/local/lib/libomp/
-CC=g++
-CFLAGS=-std=c++11 -O3 -g -Wall -fopenmp -fPIC -I/usr/include \
+EIGEN_INCLUDE_DIR ?= /usr/include/eigen3
+CC = g++
+CFLAGS = -std=c++11 -O3 -g -Wall -fopenmp -fPIC -I/usr/include \
 	-I../include -I$(EIGEN_INCLUDE_DIR) -I$(GOOGLETEST_DIR)/include \
 	-march=native
 
-LFLAGS=-fopenmp -L$(GOOGLETEST_DIR)/lib -lgtest
+LFLAGS = -fopenmp -L$(GOOGLETEST_DIR)/lib -lgtest
 
 all: test
 

